lib/zendesk_api/collection.rb

```diff
@@ -370,13 +415,7 @@ def _all(start_page = @options["page"], bang = false, &block)
 
       while (bang ? fetch! : fetch)
         each do |resource|
-          arguments = [resource, @options["page"] || 1]
-
-          if block.arity >= 0
-            arguments = arguments.take(block.arity)
-          end
-
-          block.call(*arguments)
+          block.call(resource, @options["page"] || 1)
```

Comment:
I can't think of any reason why these two versions should behave differently -> simplifying.